scripts/update-deps.js

```diff
@@ -1,190 +1,145 @@
-  if (!latest) {
-    // If we are not updating across semver ranges than undo any updates to
-    // package.json or catalog specs since we only want the lockfile updated
-    spawn('git', [
-      'checkout',
-      'package.json',
-      'pnpm-workspace.yaml',
-      'src/*/package.json',
-    ])
-    return
+const main = async ({ values }) => {
+  const current = replaceDepsWithCatalogVersion()
+
+  const workspaceFlags = []
+
+  if ('omitWorkspace' in values) {
+    workspaceFlags.push(
+      ...execSync('pnpm -r exec npm pkg get name', {
+        encoding: 'utf-8',
+      })
+        .trim()
+        .split('\n')
+        .map(p => JSON.parse(p))
+        .filter(p => !values.omitWorkspace.includes(p))
+        .map(w => `--workspace=${w}`),
+    )
+  } else if ('workspace' in values) {
+    workspaceFlags.push(
+      ...values.workspace.map(w => `--workspace=${w}`),
+    )
+  } else {
+    workspaceFlags.push('--workspaces')
   }
-  // Otherwise only reset dep changes that are compatible with the previous spec
-  const reset = (p, pj, type) => {
-    for (const [name, spec] of Object.entries(pj[type] ?? {})) {
-      const oldSpec = oldPackages.get(p)[type][name]
-      if (oldSpec !== spec && shouldReset(oldSpec, spec)) {
-        pj[type][name] = oldSpec
-      }
-    }
+
+  if ('noRoot' in values) {
+    workspaceFlags.push('--no-root')
   }
-  for (const [p, pj] of newPackages.entries()) {
-    reset(p, pj, 'dependencies')
-    reset(p, pj, 'devDependencies')
-    writeJson(p, pj)
+
+  for (const v in nonNCUOptions) {
+    delete values[v]
   }
-}
 
-/**
- * This script will update all dependencies to either the latest in the current
- * semver range or to the `latest` tag if `--latest` is provided. This relies
- * on pnpm's `update` command and makes assumptions about our own use such as
- * enforcing use of the `^` range operator.
- * TODO: Move this code into our own update command and make it do other things
- * like only optionally update package.json specs and support for widening semver
- * ranges for specific deps or dep types.
- */
-const main = async ({ latest, force }) => {
-  const previousModified = getModified()
-  if (previousModified.length) {
-    if (!force) {
-      throw new Error(
-        `The following files can't have local changes prior to running this script.\n` +
-          `Run with \`--force\` to have this script reset the changes in those files.\n  ` +
-          previousModified.join('\n  '),
-      )
-    }
-    for (const f of previousModified) {
-      spawn('git', ['checkout', f])
-    }
+  const res = execSync(
+    [
+      'node',
+      fileURLToPath(
+        import.meta.resolve('npm-check-updates/build/cli.js'),
+      ),
+      ...workspaceFlags,
+      '--jsonUpgraded',
+      ...Object.entries(values).map(([k, v]) => `--${k}=${v}`),
+    ].join(' '),
+    { encoding: 'utf-8' },
+  )
+
+  let jsonOutput = null
+  try {
+    jsonOutput = JSON.parse(res)
+  } catch {
+    console.log(res)
```

Comment:
do we even want to `console.log` something in here? if the answer is yes, than may I suggest using `console.warn('<here is the reason>:', res)` instead?

Reply:
That was attempting to still allow for `--help` to be passed to `npm-check-updates` which will always return a string and not json.

But I reworked it to use the programmatic API and only fallback to getting the help output if `--help` is passed explicitly.